docs/reference/async-processing.mdx

```diff
@@ -0,0 +1,129 @@
+
+1. The greater the remote latency, the bigger the impact of async processing. At 2 ms round trip latency, throughput improves by 5.5x. At 1 ms round trip latency, throughput improves by 3x.
+2. The throughput with async processing enabled is stable even with varying remote latency. The processor was ultimately bottlenecked by the resources available on the client side!
+
+:::warning
+The async processor makes high round trip network latency a non-factor in achieving high throughput. However, using the async processor will generally place more load on the remote system, which in turn may get saturated, which in turn will drive *its* latencies, which will mean the throughput gains of the async processor will be below the theoretical maximum.
+
+So make sure to load test with your remote system and tune the async processor accordingly.
+:::
+
+
+## Enabling async processing
+
+Turning async processing on (or off!) is quick and easy. All it requires is one line of code and one config and you’re good to go! Make sure to check out the “Limitations” section for what is covered by the current version and what will be coming in a future release.
+
+### 1. Select and update your existing Processor
+
+Wrap the supplier for the processor you wish to enable async processing for in the appropriate wrapper class:
+
+    1. For a `.process()` with a `ProcessorSupplier`, use `AsyncProcessorSupplier#createAsyncProcessorSupplier`
+    2. For a `.processValues()` with a  `FixedKeyProcessorSupplier` , use `AsyncFixedKeyProcessorSupplier#createAsyncProcessorSupplier`
+    
+Example:
+    
+```java
+    // Before
+    kstream.processValues(
+        new MyStatefulProcessorSupplier<>(),
+        Named.as("RegularProcessor"),
+        STORE_NAME);
+    
+    // After
+    kstream.processValues(
+        createAsyncProcessorSupplier(
+            new MyStatefulProcessorSupplier<>(),
+            Named.as("AsyncProcessor"),
+            STORE_NAME);
+            
+    // note: the Named.as was modified to highlight the example,
+    // changing the processor name is not required to enable async processing
+```
+    
+### 2. Configure the thread pool
+
+Configure the thread pool size by setting the ResponsiveConfig `responsive.async.thread.pool.size` to a positive integer. Something like 10 is a good place to start, but make sure to check out the tuning section below for a more detailed discussion of how to tune and configure the Responsive Async Processor.
+
+### 3. Ensure that you implement the Processor#stores method
+
+Note that the async processing framework requires a stateful processor to connect its StateStores by implementing the `ProcessorSupplier#stores` method. Please verify that your processor supplier implements this method before passing it into the async wrapper, or switch to this method of connecting state stores if it does not.
+You will know if this is not set up correctly by an immediate `IllegalStateException` upon startup.
```

Comment:
ah I just double-checked and we have an earlier check for this that  throws a better exception:

```suggestion
You will know if this is not set up correctly by an immediate `UnsupportedOperationException` upon startup.
```

Reply:
Got it. will update.